### ISSUE TITLE
About stress conditions

### ISSUE THREAD

**Issue by ruchitkini** [JinshuaiBai/LSWR_loss_function_PINN#3]
Hello @JinshuaiBai,

Great Work! I am new to neural networks and I was wondering about the stress conditions for your code. I am studying the perforated strip plate case. You have set the stress condition, $\sigma = 1$ on the right side, and for all the other boundaries it is set as, $\sigma = 0$. Why is it so? The value of stresses at these boundaries is not zero when I check the solution. Please guide me through this.

Thanks.

**Comment by JinshuaiBai** [comment]
As you mentioned, the stress boundary conditions at the left boundaries are obviously not 0. I am pretty sure that, although Syy and Sxy as well as Sxx are predefined in the “input_info.py”, they are not used in the following training. They are provided in the “input_info.py” so that others can more conveniently use them in their own cases.

Hope this can answer your question.

**Comment by ruchitkini** [comment]
Sorry for the follow-up question. But I still don't get it, aren't these boundary condition values used to calculate the loss ( between stress value from PINN.py file and Input_info.py) as defined in the function Loss_grad? And if you say that they are not used, all of them are not used or some specific of them are being used how would I know that? 
Because ```s_r_x = np.ones((ns_l, 1)).astype(np.float32)*1``` this is the line which sets, $\sigma = 1$ and it is needed. Would you please clear up this doubt? 

Thanks

**Comment by JinshuaiBai** [comment]
Not all the boundary conditions defined in the "Input_info.py" are used. For example, in the pure bending case, the list y in the Loss_grad function contains all the pre-defined boundary conditions. However, if you check the formula for l2 (from line 73 to 79), only y[0] y[1] y[2] y[3] y[5] y[6] y[7] are used, while the y[4] is not used. This is because, referring to the basic solid mechanics, only the stress $σ_x$ on the left boundary is unknown and y[4] denotes the stress $σ_x$ on the left boundary (you can find it in the "Input_info.py").

Hope this can help you.

**Comment by ruchitkini** [comment]
Yes. I understood now after checking the lines for the losses. Just one more question, in the perforated plate problem, in the Opt.py file I did not get the implementation of loss for stresses on the arc, L3 (lines 83-84). 
If I am correct it is given as $\sum (s_{11} + s_{12})^2 + \sum (s_{12} + s_{22})^2$. Wouldn't it just be $\sum ((s_{11})^2 +(s_{22})^2+ (s_{12})^2)$ ?

**Comment by JinshuaiBai** [comment]
The current formula for L3 is correct. This loss term denotes the residual from the traction force on the arc (the boundary of the perforated hole). Since the normal and tangential directions of the position on the arc are not the same as the x and y axes of the Cartesian coordinate, coordinate transformation is required. That is the reason why you can find the variables "nx" and "ny" in the formula (from lines 81 to 82 in the “Opt.py”)

For more details on how to calculate the stress boundary condition on arbitrary geometries, you may refer to some mechanics textbooks. Also, you may also need to read some for the free stress boundary conditions. I believe those will help you to better understand the formula.

Hope this can answer your question.

**Comment by ruchitkini** [comment]
Yes understood. Thank you so much for your responses.
